rudder-core/src/main/resources/hooks.d/node-policy-generated/readme.txt

```diff
@@ -0,0 +1,45 @@
+
+Generallity about Rudder Hooks
+==============================
+
+Hooks must be executable file, typically scripts, whatever the language
+as long they are self-executable (i.e: correct #! definition). 
+
+Hooks must be name with the extension ".hook". Any other extension will 
+be ignore (which allow to put other file in these directory, like a 
+readme, for example). 
+
+Hooks are executed sequentially, in lexical order. We encourage
+you to use the patter "NN-hookname.ext.hook", with NN a number like
+"05", "20", etc. 
+
+About node-policy-generated/ hooks
+==================================
+
+When/What ?
+-----------
+
+This directory contains hooks executed after poplicies are fully 
+generated for node, but before these new policies replace the old
+one (technically, before we move /var/rudder/share/$NODEID/rules.new 
+to /var/rudder/share/$NODEID/rules).
+
+Typically, these hooks proceed to sanity checks on the new policies
+(ex: cf-promises), update permission on files, or interect with
+external services using knowledged from the generated policies 
+(ex: send node-properties JSON file to a third party service). 
+
+Parameters ?
+------------
+Hooks parameter are passed by environment variable: 
+
+- RUDDER_GENERATION_DATETIME    : generation datetime: ISO-8601 date/time that identify that policy generation start 
+- RUDDER_NODEID                 : the nodeId
+- RUDDER_NEXT_POLICIES_DIRECTORY: new policies directory (ie /var/rudder/share/$NODEID/rules.new)
+- RUDDER_AGENT_TYPE             : agent type ("cfengine-nova" or "cfengine-community")
+
```

Comment:
Would it make sense to also add an env var about `$RUDDER_HOOK_NAME`? That could then include like `node-policy-generated`. This would help if you want to use a single binary/script (due to code/function reuse without splitting it up) by symlinking it to different hook dirs. This way you'd not need to parse `$0` go get the directory that was being used to call the script...